log_downloader_service.py

```diff
@@ -0,0 +1,87 @@
+import os
+import time
+import shutil
+from pathlib import Path
+from loguru import logger
+from git import Repo
+import wmi
+
+
+REPO_PATH = Path(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))) # TODO: change
+LOGS_DIR = REPO_PATH / "logs"
+LOGS_DIR.mkdir(exist_ok=True)
+
+
+def commit_log(repo, log_file):
+    try:
+        repo.index.add([str(log_file)])
+        repo.index.commit(f"Add log file: {log_file.name}")
+        logger.info(f"Committed {log_file.name} to repository")
+    except Exception as e:
+        logger.info(f"Error committing {log_file.name}: {e}")
+
+
+def get_file_signature(file_path):
+    return file_path.name, file_path.stat().st_size
+
+
+def is_file_downloaded(log_file):
+    file_name = log_file.name
+    file_size = log_file.stat().st_size
+
+    for existing_file in LOGS_DIR.glob("*.wpilog"):
+        if (existing_file.name == file_name and
+                existing_file.stat().st_size == file_size):
+            return True
+    return False
+
+def get_usb_drives():
+    drives = set()
+    c = wmi.WMI()
```

Comment:
I'm aware that this can be done with os, however since this is a windows only application and we need to detect if the disk is removeable in my opinion WMI is the most robust and simple solution here.

Reply:
OK, I agree about the robustness.